<launch>
  <arg name="robot_name" value="tentabot"/>
  <arg name="world_frame_name" default="odom"/>

  <arg name="start_x" value="0.0"/>
  <arg name="start_y" value="0.0"/>
  <arg name="start_z" value="0.0"/>
  <arg name="start_yaw" value="$(eval 2*pi)"/>

  <arg name="resolution" value="0.15"/>
  <arg name="world_name" value="map"/>
  
  <arg name="time_limit" value="90" />
  <arg name="dt" value="0.05" />

  <arg name="max_lat_velo" value="1" />
  <arg name="max_yaw_velo" value="2" />
  
  <arg name="tyaw_cnt" value="11" />
  <arg name="tpitch_cnt" value="1" />
  <arg name="tsamp_cnt" value="20" />
  <arg name="tlen" value="3" />
  <arg name="tyaw" value="$(eval 0.3*pi)" />
  <arg name="tpitch" value="0.0" />
  <arg name="tentacle_type" value="linear" /> 
  <!--arg name="tentacle_type" value="circular" /--> 
  <arg name="pdist" value="0.4" />
  <arg name="sdist" value="1" />
  <arg name="egrid_vdim" value="0.1" />
  <arg name="egrid_vnumx" value="100" />
  <arg name="egrid_vnumy" value="100" />
  <arg name="egrid_vnumz" value="100" />

  <arg name="crash_dist" value="2.5" />
  <arg name="clear_scale" value="50" />
  <arg name="flat_scale" value="100" />
  <arg name="close_scale" value="80" />
  <arg name="smooth_scale" value="0.5" />

  <group ns="$(arg robot_name)">

    <node name="tentabot_go_mobiman" pkg="tentabot" type="tentabot_go_mobiman" output="screen">

      <param name="world_frame_name" value="$(arg world_frame_name)" />
      <param name="start_x" value="$(arg start_x)" />
      <param name="start_y" value="$(arg start_y)" />
      <param name="start_z" value="$(arg start_z)" />
      <param name="start_yaw" value="$(arg start_yaw)" />
      <param name="resolution" value="$(arg resolution)" />
      <param name="world_name" value="$(arg world_name)"/>
      <param name="time_limit" value="$(arg time_limit)" />
      <param name="dt" value="$(arg dt)" />
      <param name="max_lat_velo" value="$(arg max_lat_velo)" />
      <param name="max_yaw_velo" value="$(arg max_yaw_velo)" />
      <param name="tyaw_cnt" value="$(arg tyaw_cnt)" />
      <param name="tpitch_cnt" value="$(arg tpitch_cnt)" />
      <param name="tsamp_cnt" value="$(arg tsamp_cnt)" />
      <param name="tlen" value="$(arg tlen)" />
      <param name="tyaw" value="$(arg tyaw)" />
      <param name="tpitch" value="$(arg tpitch)" />
      <param name="tentacle_type" value="$(arg tentacle_type)" />
      <param name="pdist" value="$(arg pdist)" />
      <param name="sdist" value="$(arg sdist)" />
      <param name="egrid_vdim" value="$(arg egrid_vdim)" />
      <param name="egrid_vnumx" value="$(arg egrid_vnumx)" />
      <param name="egrid_vnumy" value="$(arg egrid_vnumy)" />
      <param name="egrid_vnumz" value="$(arg egrid_vnumz)" />
      <param name="crash_dist" value="$(arg crash_dist)" />
      <param name="clear_scale" value="$(arg clear_scale)" />
      <param name="flat_scale" value="$(arg flat_scale)" />
      <param name="close_scale" value="$(arg close_scale)" />
      <param name="smooth_scale" value="$(arg smooth_scale)" />

    </node>
  </group>
</launch>
